modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,360 @@
+    auto outputDomainData = static_cast<UInt*>(outputDomainPacket.getData());
+
+    DataPacketPtr outputPacket = DataPacketWithDomain(outputDomainPacket, outputDataDescriptor, 1);
+    OutputType* outputData = static_cast<OutputType*>(outputPacket.getData());
+
+    auto rangeSize = outputDataDescriptor.getDimensions()[0].getSize();
+    auto labels = outputDataDescriptor.getDimensions()[0].getLabels();
+    if (labels.getCount() == 0) return;
+
+    size_t packageVals = 0;
+    while (outputPackages && packets.size()) {
+        auto listPacket = packets.front();
+
+        auto sampleCnt = listPacket.getSampleCount();
+        auto inputData = static_cast<InputType*>(listPacket.getData());
+        auto inputDomainData = static_cast<UInt*>(listPacket.getDomainPacket().getData());
+       
+        // reset array for new package
+        if (packageVals == 0) {
+            memset(outputData, 0, rangeSize * sizeof(OutputType));
+        }
+
+        bool packetInEpoch = timeInInterval(packetStarted, inputDomainData[sampleCnt - 1]);
+
+        size_t sampleIdx = sampleStarted;
+        for (; sampleIdx < sampleCnt; sampleIdx++) {
+            // if there are values from another interval - check each value
+            if (!packetInEpoch) {
+                if (!timeInInterval(packetStarted, inputDomainData[sampleIdx]))
+                    break;
+            }
+
+            outputDomainData[0] = inputDomainData[sampleIdx];
+            packageVals++;
+
+            auto& rawData = inputData[sampleIdx];
+
+            // filter values
+            if (static_cast<Float>(rawData) < static_cast<Float>(labels[0]))
+                continue;
+            else if (static_cast<Float>(rawData) > static_cast<Float>(labels[rangeSize - 1]))
+                continue;
+
+            if (labels.getCount() == 1) {
+                outputData[0] += 1;
+            } else {
+                // binary search 
+                size_t low = 0;
+                size_t high = labels.getCount() - 1;
+
+                while (low <= high) {
+                    size_t mid = low + (high - low) / 2;
+
+                    if (rawData >= static_cast<Float>(labels[mid]) && rawData < static_cast<Float>(labels[mid + 1])) {
+                        outputData[mid] += 1;
+                        break;
+                    } else if (rawData < static_cast<Float>(labels[mid])) {
+                        high = mid - 1;
+                    } else {
+                        low = mid + 1;
+                    }
+                }
+            }
+        }
+
+        if (packetInEpoch) {
+            sampleStarted = 0;
+            packets.pop_front();
+        } else {
+            if (packageVals) {
+                for (size_t i = 0; i < rangeSize; i++) {
+                    outputData[i] /= packageVals;
+                }
+            }
+
+            outputSignal.sendPacket(outputPacket);
+            outputDomainSignal.sendPacket(outputDomainPacket);
```

Comment:
If `outputPackages > 1` this will create issues, as you're sending the same packet object more than once, and re-writing the data buffer of the same packet. A new packet with its own buffer needs to be created for every `sendPacket` call.